src/Serializer/AbstractItemNormalizer.php

```diff
@@ -488,4 +499,89 @@ protected function normalizeRelation(PropertyMetadata $propertyMetadata, $relate
 
         return $iri;
     }
+
+    /**
+     * Sets the object to populate if allowed to and not set already. If you pass identifiers, it will search for them,
+     * otherwise it falls back to the provided IdentifiersExtractor.
+     *
+     * @param $data
+     * @param string $class
+     * @param array  $context
+     * @param array  $identifiers
+     */
+    protected function setObjectToPopulate($data, $class, array $context = [])
+    {
+        if (isset($context[self::OBJECT_TO_POPULATE]) || !\is_array($data)) {
+            return;
+        }
+
+        $updateAllowed = isset($context['api_allow_update']) && true !== $context['api_allow_update'];
+        $identifiers = $this->getIdentifiersForDenormalization($class);
+        $writeableIdentifiers = [];
```

Comment:
this logic is flawed unfortunately. You cannot validate the writability of identifiers here, as identifiers can be used to reference nested relations, to update them. example:
https://github.com/api-platform/core/blob/master/features/hal/max_depth.feature#L58

summary. having an identifier in the data does not mean it is trying to be updated.